manifests/chain.pp

```diff
@@ -1,23 +1,33 @@
 # defined resource which creates all rules for one chain
 # @param policy Set the default policy for a CHAIN
 # @param disable_conntrack Disable/Enable usage of conntrack
-# @param chain Name of the chain that should be managed
 # @param log_dropped_packets Enable/Disable logging of packets to the kernel log, if no explicit chain matched
+# @param chain Name of the chain that should be managed
+# @param table Select the target table (filter/raw/mangle/nat)
 define ferm::chain (
   Ferm::Policies $policy,
   Boolean $disable_conntrack,
   Boolean $log_dropped_packets,
   String[1] $chain = $name,
+  Ferm::Tables $table = 'filter',
 ) {
 
+  # prevent unmanaged files due to new naming schema
+  # keep the default "filter" chains in the original location
+  # only prefix chains in other tables with the table name
+  if $table == 'filter' {
+    $filename = "${ferm::configdirectory}/chains/${chain}.conf"
+  } else {
+    $filename = "${ferm::configdirectory}/chains/${table}-${chain}.conf"
```

Comment:
at the moment, those files are not included in the [ferm.conf](https://github.com/voxpupuli/puppet-ferm/blob/master/templates/ferm.conf.epp#L18-L26).